pkg/eks/api/api.go

```diff
@@ -41,22 +42,45 @@ var DefaultWaitTimeout = 20 * time.Minute
 // DefaultNodeCount defines the default number of nodes to be created
 const DefaultNodeCount = 2
 
-// ClusterProvider provides an interface with the needed AWS APIs
+// ClusterLocator is what identifies a cluster
+type ClusterLocator struct {
+	Name   string
+	Region string
+}
+
+// String returns canonical representation of ClusterLocator
+func (c *ClusterLocator) String() string {
+	return fmt.Sprintf("%s.%s.eksctl.io", c.Name, c.Region)
+}
+
+// LogString returns representation of ClusterLocator for logs
+func (c *ClusterLocator) LogString() string {
+	return fmt.Sprintf("EKS cluster %q in %q region", c.Name, c.Region)
+}
+
+// ClusterProvider is the interface to AWS APIs
 type ClusterProvider interface {
 	CloudFormation() cloudformationiface.CloudFormationAPI
 	EKS() eksiface.EKSAPI
 	EC2() ec2iface.EC2API
 	STS() stsiface.STSAPI
+	Region() string
```

Comment:
We may need some guidance for contributors when to split out / duplicate fields like we have done with `region`.  I get why its done but its probably something we need to keep an eye on just in case there is a proliferation .

Reply:
Yes, I agree. I wanted to avoid `Profile()` here, but it's used somewhere and I didn't manage to get rid of it in a sensible manner.